src/btree/bt_read.c

```diff
@@ -184,6 +184,300 @@ __bt_reconstruct_delta(WT_SESSION_IMPL *session, WT_REF *ref, WT_ITEM *delta)
+
+    return (cmp);
+}
+
+/*
+ * __bt_merge_internal_deltas --
+ *     Perform a k-way merge sort on the delta arrays to create a single consolidated delta array.
+ */
+static int
+__bt_merge_internal_deltas(WT_SESSION_IMPL *session, WT_CELL_UNPACK_DELTA_INT **unpacked_deltas,
+  uint32_t start, uint32_t end, size_t *sizes, WT_CELL_UNPACK_DELTA_INT ***deltas_merged,
+  size_t *size)
+{
+    WT_CELL_UNPACK_DELTA_INT **left, **right, **merged;
+    size_t left_size, right_size;
+    uint32_t mid, i;
+
+    WT_ASSERT(session, start <= end);
+    left = right = merged = NULL;
+
+    if (start == end) {
+        *size = sizes[start];
+        WT_RET(__wt_calloc_def(session, *size, &merged));
+        for (i = 0; i < (uint32_t)sizes[start]; ++i)
+            merged[i] = &unpacked_deltas[start][i];
+        *deltas_merged = merged;
+        return (0);
+    }
+
+    mid = (start + end) / 2;
+
+    WT_RET(
+      __bt_merge_internal_deltas(session, unpacked_deltas, start, mid, sizes, &left, &left_size));
+    WT_RET(__bt_merge_internal_deltas(
+      session, unpacked_deltas, mid + 1, end, sizes, &right, &right_size));
+
+    WT_RET(__wt_calloc_def(session, left_size + right_size, &merged));
+
+    WT_MERGE_SORT(session, left, left_size, right, right_size, __bt_unpacked_delta_key_cmp, true,
+      merged, *size);
+    *deltas_merged = merged;
+
+    __wt_free(session, left);
+    __wt_free(session, right);
+    return (0);
+}
+
+/*
+ * __bt_build_ref --
+ *     Build ref
+ */
+static int
+__bt_build_ref(WT_SESSION_IMPL *session, WT_REF *parent_ref, WT_CELL_UNPACK_ADDR *base_key,
+  WT_CELL_UNPACK_ADDR *base_val, WT_CELL_UNPACK_DELTA_INT *delta, bool base, WT_REF **refp,
+  size_t *incrp)
+{
+    WT_REF *ref;
+    uint8_t key_type, value_type;
+
+    WT_ASSERT(session, incrp != NULL);
+
+    WT_RET(__wt_calloc_one(session, refp));
+    *incrp += sizeof(WT_REF);
+
+    ref = *refp;
+    ref->home = parent_ref->page;
+    key_type = base ? base_key->type : delta->key.type;
+    value_type = base ? base_val->type : delta->value.type;
+
+    switch (key_type) {
+    case WT_CELL_KEY:
+        if (base)
+            __wt_ref_key_onpage_set(parent_ref->page, ref, base_key);
+        else {
+            WT_RET(__wti_row_ikey(session, 0, delta->key.data, delta->key.size, ref));
+            *incrp += sizeof(WT_IKEY) + delta->key.size;
+        }
+        break;
+    case WT_CELL_KEY_OVFL:
+    /* Overflow keys are not supported. */
+    default:
+        WT_RET(__wt_illegal_value(session, delta->key.type));
+    }
+
+    switch (value_type) {
+    case WT_CELL_ADDR_INT:
+        F_SET(ref, WT_REF_FLAG_INTERNAL);
+        break;
+    case WT_CELL_ADDR_LEAF:
+    case WT_CELL_ADDR_LEAF_NO:
+        F_SET(ref, WT_REF_FLAG_LEAF);
+        break;
+    case WT_CELL_ADDR_DEL:
+    /* Fast truncated pages are not supported. */
+    default:
+        WT_RET(__wt_illegal_value(session, delta->value.type));
+    }
+
+    if (base)
+        ref->addr = base_val->cell;
```

Comment:
In the base page case, we also don't need to handle page deleted structure?

Reply:
Based on my understanding, we should never create a delta if there is a fast-truncated child ref on the disk. This means that if deltas exist, it implies there were never fast-truncated entries on the base; otherwise, a full disk image would have been created instead.

Reply:
No, there can be fast-truncated page that is not changed on the base image.

Reply:
discussed this offline, closing this thread.